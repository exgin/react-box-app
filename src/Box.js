import React from 'react';
import './Box.css';

const Box = ({ id, removeBox, color, width, height }) => {
  const remove = () => {
    removeBox(id);
  };

  return (
    <div className='Box'>
      <p style={{ backgroundColor: color, height: `${height}px`, width: `${width}px` }}>box</p>
      <button className='Box-btn' onClick={remove}>
        Remove {color}
      </button>
    </div>
  );
};

export default Box;
