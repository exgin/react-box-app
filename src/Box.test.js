import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

it('should render successfully', function () {
  render(<Box />);
});

it('should match snapshot', function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
