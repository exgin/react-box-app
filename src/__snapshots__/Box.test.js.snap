// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <p>
      box
    </p>
    <button
      class="Box-btn"
    >
      Remove 
    </button>
  </div>
</DocumentFragment>
`;
